### PR TITLE
chore: update sample/scripts/update-rn.sh to v0.70.5

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description
CI runners don't include iOS 15 by default anymore we upgraded to run e2e test on iOS 16 but the sample start failing to build.
Our sample app was getting outdated RN 0.67
Since we are keeping only one sample close to the latest release RN I upgraded it to 0.70 (not new architecture, new RN version but still the same old architecture)

## :bulb: Motivation and Context
Failing builds: https://github.com/getsentry/sentry-react-native/actions/runs/3435819187/jobs/5728625321
Closes: https://github.com/getsentry/sentry-react-native/pull/2459
Closes: https://github.com/getsentry/sentry-react-native/pull/2604

## :green_heart: How did you test it?
Unit tests. Run the sample on simulator.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

- [ ] Wait for CI to pass, this can only be merged if e2e tests pass (at the moment iOS passed 🙂, but Android failed, but it looks like test flakiness)

#skip-changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,9 @@
 - Bump Cocoa SDK from v7.29.0 to v7.30.0 ([#2597](https://github.com/getsentry/sentry-react-native/pull/2597))
   - [changelog](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7300)
   - [diff](https://github.com/getsentry/sentry-cocoa/compare/7.29.0...7.30.0)
+- Bump Sample React Native from v0.67.4 to v0.70.5 ([#2614](https://github.com/getsentry/sentry-react-native/pull/2614))
+  - [changelog](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v0705)
+  - [diff](https://github.com/facebook/react-native/compare/v0.67.4...v0.70.5)
```
Comment (marandaneto) [inline CHANGELOG.md:19]: Sample isn't a user-facing change, so this is not needed

File: sample/android/gradle.properties [hunk 2]
```diff
@@ -9,8 +9,9 @@
 
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
-# Default value: -Xmx10248m -XX:MaxPermSize=256m
-org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+
+# Default value: -Xmx512m -XX:MaxMetaspaceSize=256m
+org.gradle.jvmargs=-Xmx2048m -XX:MaxMetaspaceSize=512m
```
Comment (marandaneto) [inline sample/android/gradle.properties:14]: I'd leave it as it is, including the last 2 commands, they don't harm anything.

File: test/client.rn.before.0.63.test.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import * as RN from 'react-native';
```
Comment (marandaneto) [inline test/client.rn.before.0.63.test.ts:1]: Why do we need this test file with this name?

Comment (krystofwoldrich) [reply test/client.rn.before.0.63.test.ts:1]: [This change on fly](https://github.com/getsentry/sentry-react-native/pull/2614/files/7abdd9d8f28600dc38b8d4b8de7f015460a52e58#diff-d02780c2f062abcf8505db670090d74c454de33dcf58974ba92e68db603208a2L123) that worked with the previous jest configuration doesn't work with the new RN settings.

**Comment by marandaneto** [comment]
We'd need the other changes from https://docs.sentry.io/platforms/react-native/troubleshooting/#react-native-069-and-higher I'd assume, the `module.exports` part.

**Comment by krystofwoldrich** [comment]
> 
The sample links our library manually (besides the e2e tests), so it's not an issue.

At least from my testing, the e2e test works as it is, at this moment the `not allowed` parts (ios.sharedlibraries and dependency.hooks) are causing a warning but the Release build is linked correctly.

File: sample/android/build.gradle [hunk 4]
```diff
@@ -11,8 +11,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        // do not upgrade the AGP before https://github.com/react-native-community/upgrade-support/issues/38 gets fixed
```
Comment (krystofwoldrich) [inline sample/android/build.gradle:14]: I've read this comment and the issue still looks open.

Nevertheless, the new RN is not compatible with the older AGP.

**Comment by krystofwoldrich** [comment]
Finally, iOS also passed after 5 tries. 

Can you take one more look @marandaneto?

File: sample/ios/Podfile [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
+
   use_react_native!(
-    :path => config["reactNativePath"],
-    # to enable hermes on iOS, change `false` to `true` and then install pods
-  	:hermes_enabled => true
+    :path => config[:reactNativePath],
+    # Hermes is now enabled by default. Disable by setting this flag to false.
+    # Upcoming versions of React Native may rely on get_default_flags(), but
+    # we make it explicit here to aid in the React Native upgrade process.
+    :hermes_enabled => true,
```
Comment (marandaneto) [inline sample/ios/Podfile:15]: H: Enable it for Android as well (Gradle file).

Comment (krystofwoldrich) [reply sample/ios/Podfile:15]: Maybe we can do it as a separate PR? Because just flipping the switch doesn't work and this task block other PRs.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**